### PR TITLE
New design for Apache Arrow Flight testers

### PR DESCRIPTION
**Pull request by CGodiksen**
This PR includes a new design for the server and manager Apache Arrow Flight testers that is based on classes instead of functions. This design makes it easier to share functionality between the server and manager tester and is especially beneficial when more common functionality is added in the future such as drop table and truncate table.

### CODE REVIEW

File: Apache-Arrow-Flight-Tester/common.py [hunk 1]
```diff
@@ -1,91 +1,81 @@
...
+            ("timestamp", "TIMESTAMP"), ("power_output", "FIELD"),
+            ("wind_speed", "FIELD"), ("temperature", "FIELD(5%)")
+        ], model_table=True)
 
-    return list(response)
+        print("\nCurrent tables:")
+        table_names = self.list_table_names()
+        print(table_names)
 
+        print(f"\nSchema for {table_names[0]}:")
```
Comment (skejserjensen) [inline Apache-Arrow-Flight-Tester/common.py:69]: Maybe replace the printing of the two table names and schemas with a loop to remove the duplicate code and so the code will continue to work if test tables are added or removed.

File: Apache-Arrow-Flight-Tester/common.py [hunk 2]
```diff
@@ -1,91 +1,81 @@
...
+        self.create_table("test_model_table_1", [
+            ("location", "TAG"), ("install_year", "TAG"), ("model", "TAG"),
+            ("timestamp", "TIMESTAMP"), ("power_output", "FIELD"),
+            ("wind_speed", "FIELD"), ("temperature", "FIELD(5%)")
+        ], model_table=True)
 
-    return list(response)
+        print("\nCurrent tables:")
+        table_names = self.list_table_names()
+        print(table_names)
```
Comment (skejserjensen) [inline Apache-Arrow-Flight-Tester/common.py:67]: Maybe delete this line as the tables names and schemas as printed below.

File: Apache-Arrow-Flight-Tester/manager.py [hunk 3]
```diff
@@ -1,50 +1,62 @@
...
+        action_body = encoded_node_url + encoded_node_mode
+        response = self.do_action("RegisterNode", action_body)
 
-def remove_node(flight_client: FlightClient, node_url: str) -> list[Result]:
-    encoded_node_url = common.encode_argument(node_url)
+        return response[0].body.to_pybytes()
 
-    return common.do_action(flight_client, "RemoveNode", encoded_node_url)
+    def remove_node(self, node_url: str) -> list[Result]:
+        """Remove the node with the given URL from the manager"""
```
Comment (skejserjensen) [inline Apache-Arrow-Flight-Tester/manager.py:34]: Mising . at the end of the docstring.

File: Apache-Arrow-Flight-Tester/manager.py [hunk 4]
```diff
@@ -1,50 +1,62 @@
...
+    manager_client = ModelarDBManagerFlightClient("grpc://127.0.0.1:9998")
+    manager_client.create_test_tables()
+
+    print(manager_client.initialize_database(["test_table_1"]))
+
+    print(manager_client.register_node("grpc://127.0.0.1:9999", "edge"))
+    print(manager_client.remove_node("grpc://127.0.0.1:9999"))
 
-    execute_query(manager_client, "SELECT * FROM test_model_table_1 LIMIT 5")
+    manager_client.query("SELECT * FROM test_model_table_1 LIMIT 5")
```
Comment (skejserjensen) [inline Apache-Arrow-Flight-Tester/manager.py:62]: It seems `query()` is executed without first registering a cloud node with the manager for executing the query, if this is on purpose, maybe add a comment that documents why this is done.

**Review by skejserjensen — CHANGES_REQUESTED** [review]

File: Apache-Arrow-Flight-Tester/common.py (hunk 1 above)
Comment (CGodiksen) [reply Apache-Arrow-Flight-Tester/common.py:69]: Fixed.

File: Apache-Arrow-Flight-Tester/common.py (hunk 2 above)
Comment (CGodiksen) [reply Apache-Arrow-Flight-Tester/common.py:67]: Fixed.

File: Apache-Arrow-Flight-Tester/manager.py (hunk 3 above)
Comment (CGodiksen) [reply Apache-Arrow-Flight-Tester/manager.py:34]: Fixed.

File: Apache-Arrow-Flight-Tester/manager.py (hunk 4 above)
Comment (CGodiksen) [reply Apache-Arrow-Flight-Tester/manager.py:62]: Added text to docstring that explains that it is assumed that at least one cloud node is already registered.

**Review by skejserjensen — APPROVED** [review]

**Pull request merged**